routes/authentication.js

```diff
@@ -4,13 +4,14 @@ const User = require('../models/user');
 const Counsellor = require('../models/counsellor');
 const config = require('../config');
 
-function tokenForUser(user) {
-    const timestamp = new Date().getTime();
-    return jwt.encode({ sub: user.ID, iat: timestamp }, config.secret);
+function tokenForUser(user, role) {
```

Comment:
Are there any plans to do the actual user lookup and password hash comparison here, or is that to be implemented somewhere else?

Reply:
The user/counsellor lookup and password hash comparison is done in services/passport.js, which will be run before signin or signinCounsellor in authentication.js is run.

routes/index.js is where I placed all the routes so you'll be able to see how it's pieced together.